### PR TITLE
Add com.snowplowanalytics.snowplow.ecommerce/promotion/jsonschema/1-0-0

### LINKED ISSUE #1263: Add com.snowplowanalytics.snowplow.ecommerce/promotion/jsonschema/1-0-0
Add ecommerce promotion schema.

### PR DESCRIPTION
**Pull request by igneel64**
Add ecommerce promotions schema.

close #1263

### CODE REVIEW

File: schemas/com.snowplowanalytics.snowplow.ecommerce/promotion/jsonschema/1-0-0 [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            ],
+            "maxLength": 256
+        },
+        "type": {
+            "description": "Type of the promotion delivery mechanism. E.g. popup, banner, intra-content",
+            "type": [ "string", "null" ],
+            "maxLength": 256
+        },
+        "slot": {
+            "description": "The website slot in which the promotional content was added to.",
```
Comment (emielver) [inline schemas/com.snowplowanalytics.snowplow.ecommerce/promotion/jsonschema/1-0-0:53]: Could you maybe add some examples here as well?

File: schemas/com.snowplowanalytics.snowplow.ecommerce/promotion/jsonschema/1-0-0 [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            "type": "string",
+            "maxLength": 4096
+        },
+        "name": {
+            "description" : "The name of the promotion.",
+            "type": [ "string", "null" ],
+            "maxLength": 256
+        },
+        "products" : {
+            "description" : "Array of SKUs or product IDs showcased in the promotion.",
```
Comment (emielver) [inline schemas/com.snowplowanalytics.snowplow.ecommerce/promotion/jsonschema/1-0-0:23]: Is this going to be ordered based on the order of the products in the promotion, or unsorted?

**Review by emielver — COMMENTED** [review]
Just some questions!

File: schemas/com.snowplowanalytics.snowplow.ecommerce/promotion/jsonschema/1-0-0 [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            "description" : "The ID of the promotion.",
+            "type": "string",
+            "maxLength": 4096
+        },
+        "name": {
+            "description" : "The name of the promotion.",
+            "type": [ "string", "null" ],
+            "maxLength": 256
+        },
+        "products" : {
```
Comment (matus-tomlein) [inline schemas/com.snowplowanalytics.snowplow.ecommerce/promotion/jsonschema/1-0-0:22]: Nit: why not call this `product_ids` directly? We have the `id` field in the `product` schema, so I think naming it `product_ids` would make the relation more explicit.

File: schemas/com.snowplowanalytics.snowplow.ecommerce/promotion/jsonschema/1-0-0 [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+            "minimum": 0,
+            "maximum": 9999999
+        },
+        "creative_id": {
+            "description" : "Identifier/Name/Url for the creative presented on the promotion.",
+            "type" : [
+                "string",
+                "null"
+            ],
+            "maxLength": 256
```
Comment (matus-tomlein) [inline schemas/com.snowplowanalytics.snowplow.ecommerce/promotion/jsonschema/1-0-0:45]: If the field should accept URLs as well (as stated in description), 256 max length seems too short.

**Review by matus-tomlein — APPROVED** [review]
LGTM!

**Comment by igneel64** [comment]
Cherry picked into R140 https://github.com/snowplow/iglu-central/pull/1256

**Pull request closed without merging**